include/mesh/mesh.tcc

```diff
@@ -2134,6 +2134,10 @@ void mpm::Mesh<Tdim>::create_nodal_properties() {
                                        materials_.size());
     nodal_properties_->create_property("normal_unit_vectors", nrows,
                                        materials_.size());
+    nodal_properties_->create_property("wave_velocities", nrows,
+                                       materials_.size());
+    nodal_properties_->create_property("density", nodes_.size(),
+                                       materials_.size());
```

Comment:
You always need the "interface" boolean to be true to activate this. This will initiate all other properties other than needed to apply for the BC. That might make the simulation slower.

Reply:
Nope, not `"interface"` boolean. Its the first line in `mpm::MPMScheme<Tdim>::absorbing_boundary_properties()` function. I'm ok with this. If things are very slow we will refactor

Reply:
Don't you need to initialize the property via this function?
https://github.com/geomechanics/mpm/blob/e110a455d997d784092a5ce13acb27be08f12d7d/include/mesh/mesh.tcc#L2111-L2149
This is only called if `interface_ `is `true`
https://github.com/geomechanics/mpm/blob/e110a455d997d784092a5ce13acb27be08f12d7d/include/solvers/mpm_explicit.tcc#L121
However, by default, `interface_` is `false`
https://github.com/geomechanics/mpm/blob/e110a455d997d784092a5ce13acb27be08f12d7d/include/solvers/mpm_explicit.h#L86-L87

Reply:
As I mentioned, no interface bool required
https://github.com/geomechanics/mpm/blob/boundary/absorbing/include/solvers/mpm_scheme/mpm_scheme.tcc#L171L192

Reply:
Ah Okay! Sorry, I missed this one.

Reply:
I looked into it, is there any reason why you create the nodal properties at every time step? Cannot we make it just once at the beginning of the time step, let's say if `absorbing_boundary_ == true`?

Reply:
I think @cgeudeker was just following the existing work for contact where the nodal properties are created every loop. I just tried putting something like this

```
if(absorbing_boundary_){
  //
  mesh_->create_nodal_properties();

  // Initialise nodal properties
  mesh_->initialise_nodal_properties();
}
```

into `initialise_mesh()` and removed it from `absorbing_boundary_properties()` (i.e., "make it just once at the beginning") and the absorbing boundary didn't work. Is this what you were recommending?